webapp/components/test-runs-query-builder.html

```diff
@@ -148,6 +175,52 @@ <h3>
         this.set(`products.${i}`, product);
       }
 
+      computeSHAsURL(query) {
+        const url = new URL(`/api/shas${query || ''}`, window.location);
+        url.searchParams.delete('sha');
+        return url;
+      }
+
+      shasURLUpdated(url) {
+        fetch(url).then(r => r.json()).then(s => {
+          this.shas = s;
+        });
+      }
+
+      shasUpdated(sha, shas) {
+        if (!shas || !this.queryBuilderSHA) {
+          return;
+        }
+        if (sha) {
+          shas = shas.filter(s => s.startsWith(sha));
+        }
+        shas = shas.slice(0, 10);
+        // Check actually different from current.
+        const curr = this.shasAutocomplete || [];
+        if (curr.length === shas.length) {
+          let same = true;
+          for (let i = 0; i < shas.length && same; i++) {
+            if (i >= curr.length || shas[i] !== curr[i]) {
+              same = false;
```

Comment:
A few suggestions about this loop:

1. Why do we need to check `i >= curr.length`? The whole loop only runs if `curr` and `shas` have the same length.
2. We can simply break here, instead of checking `same` every time.
3. Once 1 & 2 are done, then we can simplify the 3-clause for into a for...of (which, unlike for...in, is guaranteed to iterate in order).

Reply:
Updated to match the versions autocomplete logic I simplified earlier today.